### ISSUE TITLE
Respect Archived repositories

### ISSUE THREAD

**Issue by dabutvin** [dabutvin/ImgBot#93]
Don't try to clone and optimize the repos that are marked as archived

This is unnecessary work

**Comment by dabutvin** [comment]
I think [here](https://github.com/dabutvin/ImgBot/blob/master/CompressImagesFunction/CompressImagesFunction.cs#L30) might be a good place to try to add the logic to check if a repo is archived.

The webhooks don't appear to contain this info so we will have to use the installationtoken to authenticate to GitHub API and check if the repo is archived.

If it is and archived repo, I think we can simply log it and return.

see https://developer.github.com/v3/repos/#get for Repos Get docs

to authenticate to GitHub follow https://github.com/dabutvin/ImgBot/blob/master/OpenPrFunction/PullRequest.cs#L12-L16

**Comment by dabutvin** [comment]
Let's do this! Should be super quick, maybe someone will pick it up for a hacktoberfest tshirt?

**Comment by mwanchap** [comment]
I'd like to have a go at this one too, although I'm not quite sure where to put a unit test? I was thinking at the bottom of ScheduleTests.cs?

**Comment by mwanchap** [comment]
Disregard my question above, I'll just add another test class for testing checks on git repos.  Might be more in the future (e.g. for issue #98 )

**Comment by mwanchap** [comment]
That was fun, I got nerd-sniped for like an hour learning about NSubstitute.  Completed in PR #191

**Comment by dabutvin** [comment]
fixed in #191 

Agreed about it being extended in the future.

I could see a setup where we get an instance of `RepoCheck` and it has more information hanging off of it.  This is what I was thinking

```csharp
var repoCheck = new RepoCheck(
            {
                Password = installationToken.Token,
                RepoName = compressImagesMessage.RepoName,
                RepoOwner = compressImagesMessage.Owner
            });

if (repoCheck.IsArchived) { // do something }
if (repoCheck.HasOpenImgBotBranch) { // do something }
... etc
```
